src/app/client/src/app/app.component.ts

```diff
@@ -174,9 +175,15 @@ export class AppComponent implements OnInit, OnDestroy {
     this.appId = this.userService.appId;
   }
 
+  handleLocatoionPopupInSignup() {
+    this.hideLocatoionPopupInSignup = _.get(this.activatedRoute, 'snapshot.firstChild.firstChild.data.hideLocatoionPopup') ||
+          _.get(this.activatedRoute, 'snapshot.firstChild.firstChild.firstChild.data.hideLocatoionPopup');
+  }
+
   isLocationStatusRequired() {
     const url = this.router.url;
-    return !!(_.includes(url, 'signup') || _.includes(url, 'recover') || _.includes(url, 'sign-in'));
+    this.handleLocatoionPopupInSignup();
+    return !!(_.includes(url, 'signup') || _.includes(url, 'recover') || _.includes(url, 'sign-in') || this.hideLocatoionPopupInSignup);
```

Comment:
since it is OR condition hideLocatoionPopupInSignup will always be true for all routes hence never location popup will come for other routes.
Please check it once for other routes

Reply:
No, it will work why because `hideLocatoionPopupInSignup` is undefined in other routes its only true in signup route